lib/SyTest/Homeserver/Synapse.pm

```diff
@@ -145,16 +147,20 @@ sub start
    my $macaroon_secret_key = "secret_$port";
    my $registration_shared_secret = "reg_secret";
 
-   my $cert = $self->{paths}{cert_file} = "$cwd/keys/tls-selfsigned.crt";
-   my $key  = $self->{paths}{key_file} = "$cwd/keys/tls-selfsigned.key";
+   $self->{paths}{cert_file} = "$hs_dir/tls.crt";
+   $self->{paths}{key_file} = "$hs_dir/tls.key";
+
+   ensure_ssl_cert( $self->{paths}{cert_file}, $self->{paths}{key_file}, $bind_host );
 
    my $config_path = $self->{paths}{config} = $self->write_yaml_file( "config.yaml" => {
         server_name => $self->server_name,
         log_file => "$log",
         ( -f $log_config_file ) ? ( log_config => $log_config_file ) : (),
-        tls_certificate_path => "$cwd/keys/tls-selfsigned.crt",
-        tls_private_key_path => "$cwd/keys/tls-selfsigned.key",
-        tls_dh_params_path => "$cwd/keys/tls.dh",
+        tls_certificate_path => $self->{paths}{cert_file},
+        tls_private_key_path => $self->{paths}{key_file},
+        federation_custom_ca_list => [
+           "$cwd/keys/ca.crt",
+        ],
         use_insecure_ssl_client_just_for_testing_do_not_use => 1,
```

Comment:
Can we not turn this off then? Or what? I'm a bit confused about what has real certs and what doesn't.

Reply:
this setting controls synapse's 'simple http client', which is used for non-federation requests, including (it turns out) some to IS servers. There's no way to set a custom CA for that, hence the setting.

This should probably be in a comment in the sytest code...